### PR TITLE
Adds a test for broadcasting fields with arrays with singleton dimensions

### PR DESCRIPTION
**Pull request by glwagner**
This adds a test that should fail; but we should be able to fix it by defining 

```julia
@inline Base.Broadcast.materialize!(dest::AbstractField, bc::Broadcasted{<:DefaultArrayStyle}) =
    Base.Broadcast.materialize!(interior(dest), bc)
```

### CODE REVIEW

**Comment by francispoulin** [comment]
I tried the simple example from #1942 and that worked great with this PR, so that's good news.

Is there something in particular that you are expecting to fail? I will be keeping my fingers crossed.

**Comment by glwagner** [comment]
I expect 3 of these 4 tests to fail:

```julia
        grid = RegularRectilinearGrid(size=(2, 2, 2), extent=(1, 1, 1))

         c = CenterField(CPU(), grid)
         random_column = reshape(rand(2), 1, 1, 2)

         c .= random_column # broadcast to every horizontal column in c

         @test c[1, 1, 2:3] .== random_column
         @test c[2, 1, 2:3] .== random_column
         @test c[1, 2, 2:3] .== random_column
         @test c[2, 2, 2:3] .== random_column
```

**Comment by francispoulin** [comment]
I think you mean something like

```
@test c[1, 1, 1:2] == random_column[1, 1, :]
@test c[2, 1, 1:2] == random_column[1, 1, :]
@test c[1, 2, 1:2] == random_column[1, 1, :]
@test c[2, 2, 1:2] == random_column[1, 1, :]
```
These all pass for me.

File: test/test_broadcasting.jl [hunk 1]
```diff
@@ -62,6 +73,22 @@
...
+        #####
+
+        grid = RegularRectilinearGrid(size=(2, 2, 2), extent=(1, 1, 1))
+
+        c = CenterField(CPU(), grid)
+        random_column = reshape(rand(2), 1, 1, 2)
+
+        c .= random_column # broadcast to every horizontal column in c
+
+        @test c[1, 1, 2:3] .== random_column
```
Comment (francispoulin) [inline test/test_broadcasting.jl:88]:
```suggestion
        @test c[1, 1, 1:2] == random_column[1, 1, :]
```

File: test/test_broadcasting.jl [hunk 2]
```diff
@@ -62,6 +73,22 @@
...
+        @test c[2, 1, 2:3] .== random_column
```
Comment (francispoulin) [inline test/test_broadcasting.jl:89]:
```suggestion
        @test c[2, 1, 1:2] == random_column[1, 1, :]
```

File: test/test_broadcasting.jl [hunk 3]
```diff
@@ -62,6 +73,22 @@
...
+        @test c[1, 2, 2:3] .== random_column
```
Comment (francispoulin) [inline test/test_broadcasting.jl:90]:
```suggestion
        @test c[1, 2, 1:2] == random_column[1, 1, :]
```

File: test/test_broadcasting.jl [hunk 4]
```diff
@@ -62,6 +73,22 @@
...
+        @test c[2, 2, 2:3] .== random_column
```
Comment (francispoulin) [inline test/test_broadcasting.jl:91]:
```suggestion
        @test c[2, 2, 1:2] == random_column[1, 1, :]
```

**Review by francispoulin — APPROVED** [review]
I think if these and other tests pass then we will have resolved the issue.

**Pull request merged**